packages/gatsby-cli/src/create-cli.ts

```diff
@@ -451,7 +463,9 @@ Creating a plugin:
     .wrap(cli.terminalWidth())
     .demandCommand(1, `Pass --help to see all available commands and options.`)
     .strict()
-    .fail((msg, err, yargs) => {
+    .fail((msg: string, _err: Error, yargs: any): void => {
+      // FIXME: could not find any mention of getCommands in the typings or the docs
+      // It might be some private API which should not be used
```

Comment:
Right, this is apparently private API. Which is a shame, because afaik there is no other way to get this list without maintaining list of commands ourselves separately.

Do you know if we can create such a list and then enforce that each command wi create with `.command` function is using one of those? (so the list doesn't get out of sync because typechecker would complain about it)

Reply:
> create such a list and then enforce that each command wi create with .command function is using one of those
Not sure, but worth trying 👍